librz/core/rtr_http.c

```diff
@@ -127,7 +127,7 @@ static int rz_core_rtr_http_run(RzCore *core, int launch, int browse, const char
 		so.accept_timeout = 1;
 	}
 
-	origcfg = core->config;
+	origcfg = rz_config_clone(core->config);
```

Comment:
i think this is creating a memleak
the original allocated structure held by core->config always gets overwritten by a new pointer.

Reply:
isn't cloning `core->config` necessary to store the original configs?

Reply:
yes, but the issue is that the whole code is very messy. i strongly suggest to use instead `rz_config_hold_*` functions and remove this replacement of the `core->config`.

essentially what you would do is allocate a new `RzConfigHold` and then call `rz_config_hold_s|i(hold, "cfg.color", ...)` for each config variable that needs to be restored later, then at the end you will just call `rz_config_hold_restore` and `rz_config_hold_free` to restore the settings and free the allocated `RzConfigHold` structure.

As pure example on how to implement this, i suggest this function: https://github.com/rizinorg/rizin/blob/dev/librz/core/cmd/cmd_print.c#L5986

Reply:
Thanks, it was really helpful.

Reply:
Made the changes